bus-mapping/src/evm/opcodes/returndatacopy.rs

```diff
@@ -27,18 +27,21 @@ impl Opcode for Returndatacopy {
         let memory = &mut call_ctx.memory;
         let length = size.as_usize();
         if length != 0 {
+            let minimal_length = dest_offset.as_usize() + length;
+            memory.extend_at_least(minimal_length);
+
             let mem_starts = dest_offset.as_usize();
             let mem_ends = mem_starts + length;
             let data_starts = offset.as_usize();
             let data_ends = data_starts + length;
-            let minimal_length = dest_offset.as_usize() + length;
+
             if data_ends <= return_data.len() {
-                memory.extend_at_least(minimal_length);
                 memory[mem_starts..mem_ends].copy_from_slice(&return_data[data_starts..data_ends]);
-            } else {
-                assert_eq!(geth_steps.len(), 1);
```

Comment:
@han0110 Given this results in a out-of-gas condition,  shouldn't this error here be handled correctly?

Reply:
I think this branch should never be entered, if this step is an exception if should be caught here:

https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/d098f5fc40f6d86c1f8ecdd527d306e1bde6e755/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L974

Reply:
Ok. I was def. hitting this assert but I don't remember the details of it. I will test this again 👌

Reply:
[here we go](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/874)